### PR TITLE
feat/type names

### LINKED ISSUE #117: Documentation should contain a mapping from OAS format to language type
It should link to [this](https://spec.openapis.org/registry/format/) and document the mapping of OAS format to language type.

### PR DESCRIPTION
**Pull request by baywet**
fixes #117

### CODE REVIEW

**Comment by ricardoboss** [comment]
Do you already want to add a column for Dart? Or only after it gets merged?

File: OpenAPI/kiota/serialization.md [hunk 1]
```diff
@@ -188,6 +188,29 @@ const deserializedUser = deserializeFromJson(result, createUserFromDiscriminator
...
+| string | [date-time](https://spec.openapis.org/registry/format/date-time.html) | [DateTimeOffset](/dotnet/api/system.datetimeoffset) | [time](https://pkg.go.dev/time#Time) | [OffsetDateTime](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html) | [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) | [DateTime](https://www.php.net/manual/en/class.datetime.php) | [datetime](https://docs.python.org/3/library/datetime.html) | |
+| string | [date](https://spec.openapis.org/registry/format/date.html) | [Kiota Date](https://github.com/microsoft/kiota-dotnet/blob/main/src/abstractions/Date.cs) | [Kiota Date Only](https://github.com/microsoft/kiota-abstractions-go/blob/main/serialization/date_only.go) | [LocalDate](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html) | [Kiota Date Only](https://github.com/microsoft/kiota-typescript/blob/main/packages/abstractions/src/dateOnly.ts) | [Kiota Date](https://github.com/microsoft/kiota-abstractions-php/blob/main/src/Types/Date.php) | [date](https://docs.python.org/3/library/datetime.html) | |
```
Comment (andrueastman) [inline OpenAPI/kiota/serialization.md:207]:
```suggestion
| string | [date](https://spec.openapis.org/registry/format/date.html) | [Kiota Date/DateOnly](https://github.com/microsoft/kiota-dotnet/blob/main/src/abstractions/Date.cs) | [Kiota Date Only](https://github.com/microsoft/kiota-abstractions-go/blob/main/serialization/date_only.go) | [LocalDate](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html) | [Kiota Date Only](https://github.com/microsoft/kiota-typescript/blob/main/packages/abstractions/src/dateOnly.ts) | [Kiota Date](https://github.com/microsoft/kiota-abstractions-php/blob/main/src/Types/Date.php) | [date](https://docs.python.org/3/library/datetime.html) | |
```

Could we make the argument that this is mapped to `DateOnly` too with the work done to use `implicit operator` in C# for net6 targets?

**Comment by baywet** [comment]
> Do you already want to add a column for Dart? Or only after it gets merged?

Only after it gets merged, but you can add the detail as another issue so we capture the work.

File: OpenAPI/kiota/serialization.md (hunk 1 above)
Comment (baywet) [reply OpenAPI/kiota/serialization.md:207]: I'm going to add it as a note below so it's clear for all types (DateOnly, TimeOnly)

Comment (baywet) [reply OpenAPI/kiota/serialization.md:207]: added

**Review by sebastienlevert — APPROVED** [review]

**Pull request merged**